### PR TITLE
Fixed the issue with backspace and delete not working in the firefox

### PR DESCRIPTION
**Pull request by NipunaMarcus**
Also added the fix for sanitation doesn't work in designer page property toggle menu due to recent class name changes

### CODE REVIEW

**Comment by NipunaMarcus** [comment]
This fix the issue where user can override the error given when there is a dashboard with same name on dashboard creation and able to create a dashboard with an existing name.

**Comment by NipunaMarcus** [comment]
Fix the issue with default dashboard view feature not working correctly and fix the issue where user can remove all the permission in dashboard setting page edit and view permissions.

**Comment by NipunaMarcus** [comment]
Fix the issue with Anon dashboard that default view is shown to the user in the same tenant but has no view/edit permissions.

**Comment by NipunaMarcus** [comment]
Fix the issue with URL where showing the query param "?isAnonView=false". After the fix this query param only will appear if user is logged in and showing the anon view

**Comment by NipunaMarcus** [comment]
Fix the issue where user is shown the page id of the landing page when user is previewing the page from dashboard designer.

**Comment by NipunaMarcus** [comment]
Fixed the issue related to the JIRA: https://wso2.org/jira/browse/UES-685

**Comment by NipunaMarcus** [comment]
Added the suggested improvement for the UI of the portal. These changes are available in the following document [1]. 

[1]. https://docs.google.com/a/wso2.com/spreadsheets/d/1i1BQtz9kShoD7FqSZ6_5UoRYJsk0Ib6Grz7JHJAoDo0/edit?usp=sharing

File: apps/portal/js/designer.js [hunk 1]
```diff
@@ -1835,69 +1835,49 @@ $(function () {
...
+                        updatePagesList(pages);
+                        initNanoScroller();
+                        return;
+                    }
+                    hideProperties();
+                    renderPage(first.id);
+                });
+            };
+
+            generateMessage("This will remove the page and all its content." +
```
Comment (sinthuja) [inline apps/portal/js/designer.js]: Use locales to get the message.

Comment (NipunaMarcus) [inline apps/portal/js/designer.js]: In js we cannot use the i18n. But this message can be loaded through a hidden element in html and showing as the message can be done.

**Pull request merged**